packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5711,23 +5711,25 @@ class RestorableRouteFuture<T> extends RestorableProperty<String?> {
   static NavigatorState _defaultNavigatorFinder(BuildContext context) => Navigator.of(context);
 }
 
-extension <T> on Iterable<T> {
-  T? firstWhereOrNull(bool Function(T element) test) {
-    for (final T element in this) {
-      if (test(element)) {
-        return element;
-      }
+/// First item satisfying the predicate, or null if not found.
+@visibleForTesting
+T? iterableFirstWhereOrNull<T>(Iterable<T> iterable, bool Function(T element) test) {
```

Comment:
just hard code this to _RouteEntry ? also maybe rename this to
_firstRouteEntryWhereOrNull

Reply:
and typedef the test method function signature

Reply:
> and typedef the test method function signature

May I know it a bit more?

```
class _RouteEntry {
  _firstRouteEntryWhereOrNull() {...}
}
```

Then I wonder how to test it using typedef

Reply:
Oh I see, you mean the argument named `test`, not "a test to the newly added function"